File: 008/prob8.py
#Project Euler Problem 8
#
#Find the greatest product of five consecutive digits in the 1000-digit number.

#opern the file and read in the number to traverse as a string
largeNumberFile = open("number.txt","r")
largeNumberString = largeNumberFile.read().replace('\n','')
greatestProduct = 0
i = 0
productCount = 0
productarray = []

while i + 4 < len(largeNumberString):
	product = 1
	
	for j in range(0, 4):
		product *= int(largeNumberString[i + j])
		productCount += 1
	
	if product > greatestProduct: greatestProduct = product
	productarray.append(product)
	i += 1

print productarray
print greatestProduct
print productCount
print i

File: 008/prob8.c
//Find the greatest product of five consecutive
//digits in the 1000-digit number.
#include <stdio.h>

int
main() {
	FILE *fp = fopen("number.txt","r");
	int a,b,c,d,e,currentProduct;
	int largestProduct = 0;

	a = fgetc(fp) - '0';
	b = fgetc(fp) - '0';
	c = fgetc(fp) - '0';
	d = fgetc(fp) - '0';
	e = fgetc(fp);

	while (e != EOF)
	{
		if (e != '\n')
		{
			e = e - '0';
			currentProduct = a*b*c*d*e;
			if (currentProduct > largestProduct) largestProduct = currentProduct;
			a = b;
			b = c;
			c = d;
			d = e;
		}

		e = fgetc(fp);
	}

	fclose(fp);

	printf("%d\n",largestProduct);

	return 0;
}
